ibis/expr/tests/test_newrels.py

```diff
@@ -1233,9 +1299,11 @@ def test_self_join_with_intermediate_selection():
         )
         assert join.op() == expected
 
-    aa = a.join(a, [a.a == a.a])["a", "b_right"]
-    aaa = aa.join(a, [aa.a == a.a])
-    with join_tables(a, a, a) as (r1, r2, r3):
+    a_ = a.view()
+    a__ = a.view()
+    aa = a.join(a_, [a.a == a_.a])["a", "b_right"]
+    aaa = aa.join(a__, [aa.a == a__.a])
+    with join_tables(aaa) as (r1, r2, r3):
```

Comment:
Hm, let's come up with some more unique variable names here 😂

Reply:
I can do a1 and a2 :)

Reply:
SGTM, thanks!